### PR TITLE
[CI] Github Actions

### PR DESCRIPTION
**Pull request by oberbichler**
Just a draft

### CODE REVIEW

**Comment by oberbichler** [comment]
image
image

**Comment by philbucher** [comment]
man this is almost too easy o,o

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: CI
+
+on: [push]
```
Comment (roigcarlo) [inline .github/workflows/build.yml]: Nice!!, but if we are to merge this please change this to [pull_request]

**Comment by armingeiser** [comment]
Awesome!

File: .github/workflows/build.yml (hunk 1 above)
Comment (oberbichler) [reply .github/workflows/build.yml]: Thx! Makes sense

**Comment by oberbichler** [comment]
- [ ] We should add Windows and...
- [ ] ...macOS.
- [x] Maybe we can use the [cache action](https://github.com/actions/cache) to speed up the building.

File: .github/workflows/build.yml (hunk 1 above)
Comment (oberbichler) [reply .github/workflows/build.yml]: BTW: Very good CI configuration script! Works like a charm...

**Comment by roigcarlo** [comment]
@oberbichler If we want to use cache we probably have to add -DCMAKE_CXX_COMPILER_LAUNCHER=ccache to the bionic script.

**Comment by oberbichler** [comment]
I think my configuration of `ccache` was/is just not correct.

**Comment by oberbichler** [comment]
Resumee after some trial-and-error-commiting:
- The problem was that the CC/CXX variables are set inside the bionic script. We can solve this problem by adding a switch/moving the variables to the CI enviromenent/adding a separate build-script/...
- The caching is working now.
- Github Actions seems to be slightly faster than Travis.
- We are currently caching ~140MB which is clearly under the limit of 500MB per cache.
- Current setup: one cache per OS. So we will not reach the repository-cache-limit of 2GB.
- I really like the combination of simplicity are versatility of the Actions.
- We should release a mobile game where you login each hour and press a "build"-button. With virtual money you can unlock features like caching, current versions of git, containers,... to reduce the waiting time. Believe me: that is a lot of fun!

**Comment by pooyan-dadvand** [comment]
Thanks @oberbichler

**Comment by roigcarlo** [comment]
@KratosMultiphysics/technical-committee really appreciates the effort you made and has no objection in approving this PR. Meanywhile, we have some suggestions for further improvements:

1. Is it posible to divide the job in following two stages: Build, Test? (and in a future Deploy which creates the wheels and deploy it to our servers)
2. Is it possible to use only one cache for all branches? In our experience with GitLab and Travis on cache for master was the most efficient.

Again, Thanks @oberbichler for your effort and initiative.

**Comment by oberbichler** [comment]
Nice that it has a benefit :)

We still need a solution for the `CC` \ `CXX` setting.

Ad 1: That should not be a problem. I have a lot of repositories on my personal GitHub page where I am building and uploading wheels. I am using `release` as a trigger but actually every trigger-type is fine. Just keep in mind that if you upload the wheels after each `pull_request` you will end up with a lot of packages on pypi.

Ad 2: That should be the current behaivor of the action. The Cache-Action has the property `key` which is specified as a string. The important thing is the usage of `-` as a separator. E.g. if there is no existing cache with key `linux-ccache-%commit-sha%` it will use the youngest cache that starts with `linux-ccache-`. It is not a perfect solution but I think there is already a discussion about that in the repo of the cache action.

We should keep in mind that the Actions are a very young feature. Im using them since the beta-phase and they were always very stable. But I strongly assume that there are still problems and there will be improvements/changes.

**Review by philbucher — DISMISSED** [review]
Thanks @oberbichler !

**Comment by oberbichler** [comment]
Seems to work.

I would suggest to add other Actions as separate files (e.g. `ci_windows.yml`)

Btw: It seems not to be necessary to set the compiler variables explicitly for travis. I added them anyway.

**Review by philbucher — DISMISSED** [review]
:+1:

**Comment by oberbichler** [comment]
@philbucher changed my mind...

**Review by philbucher — APPROVED** [review]

**Pull request merged**